core/src/system_monitor_service.rs

```diff
@@ -690,6 +694,112 @@ impl SystemMonitorService {
         })
     }
 
+    /*
+    fn string_append_u32_as_char(&mut string: String, val: u32)
+    {
+        let mut v = val;
+
+        for _i in 0..4 {
+            string.push(v as u8 as char);
+            v >>= 8;
+        };
+    }*/
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "x86_64"))]
+    unsafe fn report_cpuid_values() {
+        let cpuid_0 = __cpuid(0);
+
+        let max_leaf = cpuid_0.eax;
+        let max_subleaf_7 = if 7 <= max_leaf {
+            __get_cpuid_max(7).1
+        } else {
+            0
+        };
+
+        let mut manufacturer = String::from("");
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ebx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.edx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ecx >> (i * 8)) as u8 as char);
+        }
+
+        let cpuid_1 = if 1 <= max_leaf {
+            __cpuid(1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_2 = if 2 <= max_leaf {
+            __cpuid(2)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_4 = if 4 <= max_leaf {
+            __cpuid(4)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_0 = if 7 <= max_leaf {
+            __cpuid_count(7, 0)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_1 = if 7 <= max_leaf && 1 <= max_subleaf_7 {
+            __cpuid_count(7, 1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+
+        datapoint_info!(
+            "cpuid-values",
+            ("manufacturer_id", manufacturer, String),
+            ("cpuid_1_eax", cpuid_1.eax as i64, i64),
```

Comment:
are there any standardized, friendly names for the cpuid registers?

Reply:
I've never seen any. Always eax, etc.

Reply:
I added some names to things, which immediately made them long enough that "cargo fmt" changed the formatting and now each line of that macro is 5 lines. I really don't much enjoy cargo fmt.

Reply:
I'd honestly rather have five line metric definitions than indecipherable metric names that i have to constantly look up

Reply:
Current names are fine, imho..  I don't think expanding "cache" to "cache_and_tlb" or "extended" to "extended features" is likely to add any utility. Using these values in any way will still require looking up some bit definitions.  (unless someone would like to argue otherwise)